test/EndToEndSpec.hs

```diff
@@ -0,0 +1,38 @@
+module EndToEndSpec (spec) where
+
+import           Test.Hspec                                    (Spec, it, shouldBe)
+import           Data.Text                                     (Text)
+import qualified Data.Aeson                                    as JSON
+import qualified Data.Aeson.Types                              as JSON.T
+import qualified Language.PureScript.CoreFn.FromJSON           as CoreFn
+import           Language.PureScript.CoreFn.Module             (Module)
+import           Language.PureScript.CoreFn.Ann                (Ann)
+import           Language.PureScript.Scheme.CodeGen.Transpiler (moduleToScheme)
+import           Language.PureScript.Scheme.CodeGen.Printer    (printScheme)
+import           Language.PureScript.Scheme.CodeGen.Optimizer  (runOptimizations)
+
+readJSONFile :: FilePath -> IO (Module Ann)
+readJSONFile path = do
+  r <- JSON.decodeFileStrict' path
+  let res = r >>= JSON.T.parseMaybe CoreFn.moduleFromJSON
+  case res of
+    Nothing -> error "Nope"
+    Just (_version, module_) -> return module_
+
+compile :: FilePath -> IO Text
+compile path = (printScheme . runOptimizations . moduleToScheme)
+           <$> readJSONFile path
+
+spec :: Spec
+spec = do
+  it "parse Literals" $ do
+    res <- compile "test/purs/PureScmTest.Literals.corefn.json"
+    res `shouldBe` "(define integerLiteral2 42)\n\n(define integerLiteral1 23)\n\n(define integerArray2 (vector 23 42))\n\n(define integerArray1 (vector 1 2 3))"
```

Comment:
We should store these on files - we could have a `test/fixtures` folder where we store these, and read them in with [`readTextFile`](https://www.stackage.org/haddock/lts-17.11/turtle-1.5.20/Turtle-Prelude.html#v:readTextFile) from `turtle`.

Of course it would be nice to define a `readFixture` that reads from the right fixture folder so you don't always have to specify the prefix path